### PR TITLE
SGW Wix migration

### PR DESCRIPTION
**Pull request by kimtrang**

### CODE REVIEW

File: build/windows/wix_installer/create-installer.bat [hunk 1]
```diff
@@ -0,0 +1,36 @@
+setlocal
+set installdir=%1
+set version=%2
+set edition=%3
+set productname=%4
+set service_dir=%5
+
+if "%productname%"=="" set productname="SGW"
```
Comment (ceejatec) [inline build/windows/wix_installer/create-installer.bat]: sync-gateway

File: build/windows/wix_installer/HyperlinkExitDialog.wxs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the Microsoft Reciprocal License. See LICENSE.TXT file in the project root for full license information. -->
```
Comment (ceejatec) [inline build/windows/wix_installer/HyperlinkExitDialog.wxs]: Remove file

File: build/windows/wix_installer/sync-gateway.wxs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      <!-- Additional directory references for the Desktop and Start Menu folders -->
+      <Directory Id="DesktopFolder" Name="Desktop"/>
+    </Directory>
+
+    <Feature Id="Complete" Level="1" Title="Couchbase Server" Description="The Complete Package."
+        ConfigurableDirectory="INSTALLDIR">
+      <!-- All component files from Files.wxs -->
+      <ComponentGroupRef Id="SyncGateway"/>
+      <!-- All component files from Util.wxs -->
+      <!--<ComponentGroupRef Id="InstallerUtil"/> -->
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: remove comment

File: build/windows/wix_installer/sync-gateway.wxs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        </Directory>
+      </Directory>
+      </Directory>
+    </DirectoryRef>
+
+    <!-- See MB-24192. -->
+    <Property Id="MSIRESTARTMANAGERCONTROL" Value="Disable"/>
+
+    <InstallExecuteSequence>
+      <!-- Actions to do when REMOVE set, ie. Uninstall or Major Upgrade -->
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: Remove comment

File: build/windows/wix_installer/Config.wxi [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+<Include>
+  <?ifndef Company ?>
+    <?define Company = "Couchbase"?>
+  <?endif?>
+  <?ifndef ProductName ?>
+    <?define ProductName = "sync-gateway"?>
+  <?endif?>
+  <?if $(var.ProductName) = "sync-gateway"  ?>
+    <?define Name = "SyncGateway" ?>
+    <?define DisplayName = "Couchbase Sync Gateway" ?>
```
Comment (ceejatec) [inline build/windows/wix_installer/Config.wxi]: Just "Sync Gateway"

File: build/windows/wix_installer/Config.wxi [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    <?define Description = "Couchbase Sync Gateway mobile application REST gateway service." ?>
+    <?define UpgradeCodeID = "5E6FC699-4DCB-C674-3991-0B7EFC1B3FFD" ?>
+    <?define WinServiceApp = "sg-windows.exe" ?>
+    <?define WinSGWEXE = "sync_gateway.exe" ?>
+    <?define WinServiceConfig = "serviceconfig.json" ?>
+    <?define WinServiceControl = "auto" ?>
+  <?else?>
+    <?define Name = "SGAccel" ?>
+    <?define DisplayName = "Couchbase Sync Gateway Accelerator" ?>
```
Comment (ceejatec) [inline build/windows/wix_installer/Config.wxi]: Just "Sync Gateway Accelerator"

File: build/windows/wix_installer/sync-gateway.wxs [hunk 7]
```diff
@@ -0,0 +1,146 @@
+<?xml version='1.0' encoding='Windows-1252'?>
+<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi'
+     xmlns:util='http://schemas.microsoft.com/wix/UtilExtension'>
+
+  <?include $(sys.CURRENTDIR)\Config.wxi?>
+  <Product Id="*" Name="$(var.DisplayName)" Version="$(var.Version)" Manufacturer="Couchbase Inc."
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: Name="$(var.Company) $(var.DisplayName)"

File: build/windows/wix_installer/sync-gateway.wxs [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    <DirectoryRef Id="ServiceSGWComponent">
+        <Component Id="SGW_WindowService" Guid="68a8298e-2116-4bd0-95ec-938c07dfdb4d">
+          <File Id="sg_service_exe"
+                Name="$(var.WinServiceApp)"
+                KeyPath="yes"
+                Source="$(var.WinServiceApp)"/>
+          <ServiceInstall
+                   Id="install_SGWService"
+                   Name="$(var.Name)"
+                   DisplayName="$(var.DisplayName)"
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: add var.Company

File: build/windows/wix_installer/sync-gateway.wxs [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    <Icon Id="CouchbaseIcon" SourceFile="$(sys.CURRENTDIR)\assets\couchbase.ico"/>
+    <Property Id="ARPPRODUCTICON" Value="CouchbaseIcon" />
+    <Property Id="ARPURLINFOABOUT" Value="http://www.couchbase.com/"/>
+
+    <!-- Graphics for installer itself -->
+    <WixVariable Id="WixUIDialogBmp" Value="$(sys.CURRENTDIR)\assets\cb_installer_bg.jpg"/>
+    <WixVariable Id="WixUILicenseRtf" Value="License.rtf"/>
+
+    <!-- Text message for exit dialog -->
+    <Property Id="WIXUI_EXITDIALOGOPTIONALTEXT" Value="$(var.DisplayName) is installed. Visit http://127.0.0.1:4985/ to configure the service. Please note it may take a few moments for the console to become available." />
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: var.Company

File: build/windows/wix_installer/sync-gateway.wxs [hunk 10]
```diff
@@ -0,0 +1,146 @@
+<?xml version='1.0' encoding='Windows-1252'?>
+<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi'
+     xmlns:util='http://schemas.microsoft.com/wix/UtilExtension'>
+
+  <?include $(sys.CURRENTDIR)\Config.wxi?>
+  <Product Id="*" Name="$(var.DisplayName)" Version="$(var.Version)" Manufacturer="Couchbase Inc."
+      UpgradeCode="$(var.UpgradeCodeID)"
+      Language="1033" Codepage="1252">
+
+    <Package Id="*" Keywords="Installer" Description="$(var.Company) $(var.ProductName) $(var.Version) Installer"
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: s/ProductName/DisplayName/

File: build/windows/wix_installer/sync-gateway.wxs [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        Comments="Couchbase is a registered trademark of Couchbase Inc."
+        Manufacturer="Couchbase Inc."
+        Platform="x64"
+        InstallerVersion="500" Languages="1033" Compressed="yes" SummaryCodepage="1252"/>
+
+    <Media Id="1" Cabinet="Couchbase.cab" EmbedCab="yes" DiskPrompt="Only file"/>
+    <Property Id="DiskPrompt" Value="$(var.Company) $(var.ProductName) $(var.Version) Installation [1]"/>
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: ProductName/DisplayName

File: build/windows/wix_installer/sync-gateway.wxs [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+
+    <!-- Enable Windows Installer "major upgrade", the uninstall/install kind. -->
+    <!-- The Schedule attribute specifies where RemoveExistingProducts will go; -->
+    <!-- see https://msdn.microsoft.com/library/aa371197.aspx for details. -->
+    <!-- afterInstallExecute seems to be most efficient and also (probably?) -->
+    <!-- allows for user-modified config files to remain. -->
+    <MajorUpgrade Schedule="afterInstallExecute"
+        DowngradeErrorMessage="A later version of $(var.Company) $(var.ProductName) is already installed. Setup will now exit."/>
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: ProductName/Displayname

File: build/windows/wix_installer/sync-gateway.wxs [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+    <!-- Store the installation directory so upgrades can re-use it -->
+    <SetProperty Id="ARPINSTALLLOCATION" Value="[INSTALLDIR]" After="CostFinalize" Sequence="execute"/>
+
+    <!-- Directory layout -->
+    <Directory Id="TARGETDIR" Name="SourceDir">
+      <Directory Id="ProgramFiles64Folder" Name="PFiles">
+        <Directory Id="$(var.Company)" Name="$(var.Company)">
+          <Directory Id="INSTALLDIR" Name="$(var.ProductName)">
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: DisplayName

File: build/windows/wix_installer/sync-gateway.wxs [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+      <ComponentGroupRef Id="Shortcuts"/>
+      <ComponentRef Id="CreateLogFolder"/>
+      <ComponentRef Id="SGW_WindowService"/>
+    </Feature>
+
+    <!-- Extra componentgroup for holding the shortcuts -->
+    <SetProperty Id="CONSOLEURL" Value="http://127.0.0.1:4985/" Sequence="execute" Before="CreateShortcuts"/>
+    <ComponentGroup Id="Shortcuts">
+      <Component Id="ShortCuts" Directory="INSTALLDIR" Guid="1d9bf2ba-19fc-4bc5-a393-6f421449c8e1">
+        <RegistryValue Root="HKCU" Key="Software\$(var.Company)\$(var.ProductName)" Name="shortcut" Type="integer" Value="1" KeyPath="yes"/>
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: ProductName -> DisplayName

File: build/windows/wix_installer/sync-gateway.wxs [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+        <Shortcut Directory="DesktopFolder" Id="ConsoleShortcut"
+          Name="$(var.Company) $(var.ProductName) Console"
```
Comment (ceejatec) [inline build/windows/wix_installer/sync-gateway.wxs]: ProductName -> DisplayName

File: build/windows/sgw_windows_build.bat [hunk 16]
```diff
@@ -0,0 +1,375 @@
...
+::
+::    run by Jenkins Windows build jobs:
+::
+::        sync_windows_build.bat SGW-2.0.0+
+::
+::    with required paramters:
+::
+::          branch_name   git_commit  version  build_number  Edition  platform
+::
+::    This script only supports branches 1.2.0 and older
```
Comment (ceejatec) [inline build/windows/sgw_windows_build.bat]: ?

File: build/windows/sgw_windows_build.bat [hunk 17]
```diff
@@ -0,0 +1,375 @@
...
+    move   %BIN_DIR%\sync-gateway-accel.exe %DEST_DIR%\%ACCEL_EXEC%
+    echo "..................................SG-ACCEL Success! Output is: %DEST_DIR%\%ACCEL_EXEC%"
+
+:skip_build_sg_accel
+
+echo ======== remove build meta-data ==============
+move  %TEMPLATE_FILE%.orig  %TEMPLATE_FILE%
+
+::echo ======== test ================================
+::echo ................... running unit tests
```
Comment (ceejatec) [inline build/windows/sgw_windows_build.bat]: re-enable

**Review by tleyden — APPROVED** [review]

**Pull request merged**